sdk/core/Azure.Core.TestFramework/src/MockTransport.cs

```diff
@@ -77,6 +77,8 @@ private async Task ProcessCore(HttpMessage message)
             if (!(message.Request is MockRequest request))
                 throw new InvalidOperationException("the request is not compatible with the transport");
 
+            message.Response = null;
```

Comment:
Use ClearResponse

Reply:
Can't as this is the test framework which does not have access to Core internals.